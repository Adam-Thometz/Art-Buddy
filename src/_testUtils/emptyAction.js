const EMPTY_ACTION = { type: "" }

export default EMPTY_ACTION