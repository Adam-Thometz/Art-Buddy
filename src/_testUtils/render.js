import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
