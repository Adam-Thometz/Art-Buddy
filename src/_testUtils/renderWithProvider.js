import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;