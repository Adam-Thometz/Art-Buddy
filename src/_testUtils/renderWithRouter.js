import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import '@testing-library/jest-dom';

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
  );
};

function renderWithRouter(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) => 
    MemoryRouterWithInitialRoutes({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithRouter;