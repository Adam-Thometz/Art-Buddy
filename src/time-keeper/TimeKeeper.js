import { useEffect } from 'react';
import useVisited from '_hooks/visited/useVisited';

import { useSelector, useDispatch } from 'react-redux';
import { changeCurrGame } from '_redux/_general/generalActions';

import WindowNavbar from '_components/window-nav/WindowNavbar';
import Instructions from '_components/instructions/Instructions';

import { timeKeeper } from '_data/_activities/activityList';

const TimeKeeper = () => {
  const [hasVisited, setHasVisited] = useVisited(timeKeeper.lsKey);
  const { currGame } = useSelector(state => state.mainSettings);
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(changeCurrGame(timeKeeper));
    return () => dispatch(changeCurrGame({}));
  });

  return (<>
    <WindowNavbar page={currGame.name} />
    {!hasVisited ? <Instructions setVisited={setHasVisited} /> : <h2>00:00:00</h2>}
  </>);
};

export default TimeKeeper;