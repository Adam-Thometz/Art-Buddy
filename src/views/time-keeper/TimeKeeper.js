import { useEffect } from "react";
import useVisited from "_hooks/visited/useVisited";

import { useSelector, useDispatch } from "react-redux";
import { changeCurrGame } from "_redux/_general/generalReducer";
import { clearSong } from "_redux/time-keeper/timeKeeperReducer";

import WindowNavbar from "_components/window-nav/WindowNavbar";
import Instructions from "_components/instructions/Instructions";
import TimeLeft from "./time/time-left/TimeLeft";
import TimeBar from "./time/time-bar/TimeBar";
import Interface from "./interface/Interface";

import { timeKeeper } from "_data/_activities/activityList";

const TimeKeeper = () => {
  const [hasVisited, setHasVisited] = useVisited(timeKeeper.lsKey);
  const { currGame } = useSelector((state) => state.general);
  const { currSongIdx } = useSelector((state) => state.timeKeeper);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(changeCurrGame(timeKeeper));
    return () => dispatch(changeCurrGame({}));
  }, [dispatch]);

  const handleClear = () => {
    if (currSongIdx === 0) {
      const audio = document.querySelector(".TimeKeeperNav-audio");
      if (audio) audio.currentTime = 0;
    }
    dispatch(clearSong());
  }

  return (
    <>
      <WindowNavbar page={currGame.name} />
      {!hasVisited ? (
        <Instructions setVisited={setHasVisited} />
      ) : (
        <>
          <TimeLeft />
          <TimeBar />
          <Interface />
          <span onClick={handleClear}>clear song</span>
        </>
      )}
    </>
  );
};

export default TimeKeeper;
