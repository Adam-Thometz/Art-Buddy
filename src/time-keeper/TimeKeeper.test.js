import TimeKeeper from "./TimeKeeper";

import { render } from "_testUtils/render";

describe("TimeKeeper component", () => {
  it("renders without crashing", () => {
    render(<TimeKeeper />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<TimeKeeper />);
    expect(asFragment()).toMatchSnapshot();
  });
});
