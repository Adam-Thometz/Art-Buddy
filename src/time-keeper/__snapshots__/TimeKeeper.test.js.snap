// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeKeeper component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="WindowNavbar-games"
  >
    <h1
      class="WindowNavbar-label"
    >
      TIME KEEPER
    </h1>
  </header>
  <div
    class="Instructions"
  >
    <article>
      Tell the time with music! Add different music blocks to create a song out of the timer
    </article>
    <figure
      class="Icon"
      data-testid=""
    >
      <img
        alt=""
        src="https://art-buddy.s3.amazonaws.com/_general/play-action-btn.png"
        style="height: 250px; width: 250px; cursor: pointer;"
      />
      <figcaption
        class="Icon-text"
        style="font-size: 1.75rem;"
      >
        PLAY
      </figcaption>
    </figure>
  </div>
</DocumentFragment>
`;
